Import alias fails after being re-exported from another namespace via an export declaration when only import aliases are exported

# Bug Report

I found the using `import Something = Namespace.Something` then re-exporting that from a namespace leads to some strange behaviour, but only when import alises are the only things exported.

### 🔎 Search Terms

"Cannot use namespace" as value

### 🕗 Version & Regression Information

- This probably has existed for a while
- Tested in 5.2.0-dev.20230529

### ⏯ Playground Link

https://www.typescriptlang.org/play?ssl=24&ssc=86&pln=2&pc=3#code/CYUwxgNghgTiAEBbA9sArhBAXEBnL8A3gLABQ8SqGCADlGANYCMRZFFYyAdvvDvkwBc8fDACWXAOYBuNu048C-LAGZhXNIgBGIGLPLsQADxrIYBQnzxYmAGiv4V8AL76KzuSnSZ4dRgCZWAwoxRFNzBxt4AF5femYAOmUmN3ZQ8KVrJ1i-ROUVVIpjDKJIu0inVzkPAy9qOMYnEmD4dLNMgRiGhn8k6xS5ELD2iq7c3vzC+GKRy2T7fJdUmrk2iNyAURN2lhz4pj6BVLWCTe3zQL2Aw5tj4fX4rYzs7pUbgYMT7qf2gBYx+ITfr6OQKXj5YRYACeNBAyAAZt9zqpUjMIpYzhkWFBcGV7Jj2oEcZF-AssvjHsj-sTlP8qqRnEA

### 💻 Code

```ts
declare module pack1 {
  const test1: string;
  const test3: number;
  export { test1, test3 };
}
declare module pack2 {
  import test1 = pack1.test1;
  import test3 = pack1.test3;
  export { test1, test3 };
}
declare module pack3 {
  import test1 = pack2.test1;
  import test3 = pack2.test3;
  export { test1, test3 };
}

import packExport1 = pack1.test1;
import packExport2 = pack2.test1;
import packExport3 = pack3.test1;
import packExport4 = pack2.test1;

declare const test3: typeof packExport3;
export { packExport1 as test1, packExport2 as test2, test3, packExport4 as test4 };
```

### 🙁 Actual behavior

The types are inferred correctly, but the type checker errors:

```
> npx tsc test.d.ts
test.d.ts:12:18 - error TS2708: Cannot use namespace 'pack2' as a value.

12   import test1 = pack2.test1;
                    ~~~~~

test.d.ts:13:18 - error TS2708: Cannot use namespace 'pack2' as a value.

13   import test3 = pack2.test3;
                    ~~~~~

test.d.ts:18:22 - error TS2708: Cannot use namespace 'pack2' as a value.

18 import packExport2 = pack2.test1;
                        ~~~~~


Found 3 errors in the same file, starting at: test.d.ts:12
```

### 🙂 Expected behavior

No errors.

### Workaround

Add some other declaration and export it from any namespace that only exports import aliases:

```ts
// no diagnostics, everything is fine now
declare namespace pack1 {
  const test1: string;
  const test3: number;
  export { test1, test3 };
}
declare namespace pack2 {
  import test1 = pack1.test1;
  import test3 = pack1.test3;
  export const __dummyExport: string; // edit: actually, i discovered this doesn't need to be exported
  export { test1, test3 };
}
declare namespace pack3 {
  import test1 = pack2.test1;
  import test3 = pack2.test3;
  export const __dummyExport: string;
  export { test1, test3 };
}

import packExport1 = pack1.test1;
import packExport2 = pack2.test1;
import packExport3 = pack3.test1;
import packExport4 = pack3.test3;

declare const test3: typeof packExport3;
export { packExport1 as test1, packExport2 as test2, test3, packExport4 as test4 };
```

### Workaround 2

Use `export import`:

```ts
// no errors
declare namespace pack1 {
  const test1: string;
  const test3: number;
  export { test1, test3 };
}
declare namespace pack2 {
  export import test1 = pack1.test1;
  export import test3 = pack1.test3;
}
declare namespace pack3 {
  export import test1 = pack2.test1;
  export import test3 = pack2.test3;
}

import packExport1 = pack1.test1;
import packExport2 = pack2.test1;
import packExport3 = pack3.test1;
import packExport4 = pack3.test3;

declare const test3: typeof packExport3;
export { packExport1 as test1, packExport2 as test2, test3, packExport4 as test4 };
```

--- a/src/compiler/binder.ts
+++ b/src/compiler/binder.ts
@@ -313,7 +313,7 @@ import {
     unusedLabelIsError,
     VariableDeclaration,
     WhileStatement,
-    WithStatement,
+    WithStatement
 } from "./_namespaces/ts";
 import * as performance from "./_namespaces/ts.performance";
 
@@ -444,6 +444,14 @@ function getModuleInstanceStateForAliasTarget(specifier: ExportSpecifier, visite
                     }
                     if (found === ModuleInstanceState.Instantiated) {
                         return found;
+                    }
+                    if (statement.kind === SyntaxKind.ImportEqualsDeclaration) {
+                        // Treat re-exports of import aliases as instantiated,
+                        // since they're ambiguous. This is consistent with
+                        // `export import x = mod.x` being treated as instantiated:
+                        //   import x = mod.x;
+                        //   export { x };
+                        found = ModuleInstanceState.Instantiated;
                     }
                 }
             }

--- a/tests/baselines/reference/reExportAliasMakesInstantiated.js
+++ b/tests/baselines/reference/reExportAliasMakesInstantiated.js
@@ -1,0 +1,30 @@
+//// [tests/cases/conformance/internalModules/moduleDeclarations/reExportAliasMakesInstantiated.ts] ////
+
+//// [reExportAliasMakesInstantiated.ts]
+declare module pack1 {
+  const test1: string;
+  export { test1 };
+}
+declare module pack2 {
+  import test1 = pack1.test1;
+  export { test1 };
+}
+export import test1 = pack2.test1;
+
+declare module mod1 {
+  type test1 = string;
+  export { test1 };
+}
+declare module mod2 {
+  import test1 = mod1.test1;
+  export { test1 };
+}
+const test2 = mod2; // Possible false positive instantiation, but ok
+
+
+//// [reExportAliasMakesInstantiated.js]
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.test1 = void 0;
+exports.test1 = pack2.test1;
+var test2 = mod2; // Possible false positive instantiation, but ok

--- a/tests/baselines/reference/reExportAliasMakesInstantiated.symbols
+++ b/tests/baselines/reference/reExportAliasMakesInstantiated.symbols
@@ -1,0 +1,52 @@
+//// [tests/cases/conformance/internalModules/moduleDeclarations/reExportAliasMakesInstantiated.ts] ////
+
+=== reExportAliasMakesInstantiated.ts ===
+declare module pack1 {
+>pack1 : Symbol(pack1, Decl(reExportAliasMakesInstantiated.ts, 0, 0))
+
+  const test1: string;
+>test1 : Symbol(test1, Decl(reExportAliasMakesInstantiated.ts, 1, 7))
+
+  export { test1 };
+>test1 : Symbol(test1, Decl(reExportAliasMakesInstantiated.ts, 2, 10))
+}
+declare module pack2 {
+>pack2 : Symbol(pack2, Decl(reExportAliasMakesInstantiated.ts, 3, 1))
+
+  import test1 = pack1.test1;
+>test1 : Symbol(test1, Decl(reExportAliasMakesInstantiated.ts, 4, 22))
+>pack1 : Symbol(pack1, Decl(reExportAliasMakesInstantiated.ts, 0, 0))
+>test1 : Symbol(pack1.test1, Decl(reExportAliasMakesInstantiated.ts, 2, 10))
+
+  export { test1 };
+>test1 : Symbol(test1, Decl(reExportAliasMakesInstantiated.ts, 6, 10))
+}
+export import test1 = pack2.test1;
+>test1 : Symbol(test1, Decl(reExportAliasMakesInstantiated.ts, 7, 1))
+>pack2 : Symbol(pack2, Decl(reExportAliasMakesInstantiated.ts, 3, 1))
+>test1 : Symbol(pack2.test1, Decl(reExportAliasMakesInstantiated.ts, 6, 10))
+
+declare module mod1 {
+>mod1 : Symbol(mod1, Decl(reExportAliasMakesInstantiated.ts, 8, 34))
+
+  type test1 = string;
+>test1 : Symbol(test1, Decl(reExportAliasMakesInstantiated.ts, 10, 21))
+
+  export { test1 };
+>test1 : Symbol(test1, Decl(reExportAliasMakesInstantiated.ts, 12, 10))
+}
+declare module mod2 {
+>mod2 : Symbol(mod2, Decl(reExportAliasMakesInstantiated.ts, 13, 1))
+
+  import test1 = mod1.test1;
+>test1 : Symbol(test1, Decl(reExportAliasMakesInstantiated.ts, 14, 21))
+>mod1 : Symbol(mod1, Decl(reExportAliasMakesInstantiated.ts, 8, 34))
+>test1 : Symbol(mod1.test1, Decl(reExportAliasMakesInstantiated.ts, 12, 10))
+
+  export { test1 };
+>test1 : Symbol(test1, Decl(reExportAliasMakesInstantiated.ts, 16, 10))
+}
+const test2 = mod2; // Possible false positive instantiation, but ok
+>test2 : Symbol(test2, Decl(reExportAliasMakesInstantiated.ts, 18, 5))
+>mod2 : Symbol(mod2, Decl(reExportAliasMakesInstantiated.ts, 13, 1))
+

--- a/tests/baselines/reference/reExportAliasMakesInstantiated.types
+++ b/tests/baselines/reference/reExportAliasMakesInstantiated.types
@@ -1,0 +1,50 @@
+//// [tests/cases/conformance/internalModules/moduleDeclarations/reExportAliasMakesInstantiated.ts] ////
+
+=== reExportAliasMakesInstantiated.ts ===
+declare module pack1 {
+>pack1 : typeof pack1
+
+  const test1: string;
+>test1 : string
+
+  export { test1 };
+>test1 : string
+}
+declare module pack2 {
+>pack2 : typeof pack2
+
+  import test1 = pack1.test1;
+>test1 : string
+>pack1 : typeof pack1
+>test1 : string
+
+  export { test1 };
+>test1 : string
+}
+export import test1 = pack2.test1;
+>test1 : string
+>pack2 : typeof pack2
+>test1 : string
+
+declare module mod1 {
+  type test1 = string;
+>test1 : string
+
+  export { test1 };
+>test1 : any
+}
+declare module mod2 {
+>mod2 : typeof mod2
+
+  import test1 = mod1.test1;
+>test1 : any
+>mod1 : any
+>test1 : string
+
+  export { test1 };
+>test1 : any
+}
+const test2 = mod2; // Possible false positive instantiation, but ok
+>test2 : typeof mod2
+>mod2 : typeof mod2
+

--- a/tests/cases/conformance/internalModules/moduleDeclarations/reExportAliasMakesInstantiated.ts
+++ b/tests/cases/conformance/internalModules/moduleDeclarations/reExportAliasMakesInstantiated.ts
@@ -1,0 +1,19 @@
+declare module pack1 {
+  const test1: string;
+  export { test1 };
+}
+declare module pack2 {
+  import test1 = pack1.test1;
+  export { test1 };
+}
+export import test1 = pack2.test1;
+
+declare module mod1 {
+  type test1 = string;
+  export { test1 };
+}
+declare module mod2 {
+  import test1 = mod1.test1;
+  export { test1 };
+}
+const test2 = mod2; // Possible false positive instantiation, but ok

